tools/js2c.py

```diff
@@ -177,119 +185,152 @@ def ReadMacros(lines):
-      split = re.split('/|\\\\', name)
-      if split[0] == 'deps':
-        split = ['internal'] + split
-      else:
-        split = split[1:]
-      name = '/'.join(split)
-
-    # if its a gypi file we're going to want it as json
-    # later on anyway, so get it out of the way now
-    if name.endswith('.gypi'):
-      # Currently only config.gypi is allowed
-      assert name == 'config.gypi'
-      lines = re.sub(r'\'true\'', 'true', lines)
-      lines = re.sub(r'\'false\'', 'false', lines)
-      lines = re.sub(r'#.*?\n', '', lines)
-      lines = re.sub(r'\'', '"', lines)
-      definition = GetDefinition('config_raw', lines)
-      definitions.append(definition)
-    else:
-      AddModule(name.split('.', 1)[0], lines)
+  for filename in source_files['.js']:
+    AddModule(filename, consts, macros, definitions, initializers)
+
+  config_def, config_size = handle_config_gypi(source_files['config.gypi'])
+  definitions.append(config_def)
 
   # Emit result
-  output = open(str(target[0]), "w")
-  output.write(
-    TEMPLATE.format(definitions=''.join(definitions),
-                    initializers=''.join(initializers)))
-  output.close()
-
-def main():
-  natives = sys.argv[1]
-  source_files = sys.argv[2:]
-  if source_files[-2] == '-t':
-    global TEMPLATE
-    TEMPLATE = source_files[-1]
-    source_files = source_files[:-2]
-  JS2C(source_files, [natives])
+  definitions = ''.join(definitions)
+  initializers = '\n  '.join(initializers)
+  out = TEMPLATE.format(definitions, initializers, config_size)
+  write_if_chaged(out, target)
+
+
+def handle_config_gypi(config_filename):
+  # if its a gypi file we're going to want it as json
+  # later on anyway, so get it out of the way now
+  config = ReadFile(config_filename)
+  config = jsonify(config)
+  config_def, config_size = GetDefinition(CONFIG_GYPI_ID, config)
+  return config_def, config_size
+
+
+def jsonify(config):
+  # 1. string comments
+  config = re.sub(r'#.*?\n', '', config)
+  # 3. normalize string literals from ' into "
+  config = re.sub('\'', '"', config)
+  # 2. turn pseudo-booleans strings into Booleans
+  config = re.sub('"true"', 'true', config)
+  config = re.sub('"false"', 'false', config)
+  return config
+
+
+def write_if_chaged(content, target):
+  if os.path.exists(target):
+    with open(target, 'rt') as existing:
+      old_content = existing.read()
+  else:
+    old_content = ''
+  if old_content == content:
+    return
+  with open(target, "wt") as output:
+    output.write(content)
+
+
+def SourceFileByExt(files_by_ext, filename):
+  """
+  :type files_by_ext: dict
+  :type filename: str
+  :rtype: dict
+  """
+  ext = os.path.splitext(filename)[-1]
+  files_by_ext.setdefault(ext, []).append(filename)
+  return files_by_ext
+
+def main(args):
+  parser = argparse.ArgumentParser(
+    description='Convert code files into `uint16_t[]`s',
+    fromfile_prefix_chars='@'
+  )
+  parser.add_argument('--target', help='output file')
```

Comment:
Why not just keep the current convention? This is actually a breaking change for this script, but in a refactoring patch?

Reply:
It's part of a dependency bug fix. `tools/js2c.py` was not declared as a dependency.
https://github.com/nodejs/node/blob/6bd301c6133371594630e1bdcb51e0743abc9eb0/node.gyp#L862-L863
adding an explicit argument for output allows a succinct gyp implementation.

Reply:
P.S. As for the semverity of this change... It's a good point and I'll think about it some more. My intuition was to consider this script as an internal tool, and since there is not change in `./configure` or `Makefile` I would consider this abstracted away.